tools/dds/rtpsrelaylib/Relay.idl

```diff
@@ -92,23 +94,32 @@ module RtpsRelay {
     unsigned long long bytes_to;
   };
 
+  @mutable
+  @autoid(HASH)
+  struct ProtocolStatistics {
+    unsigned long messages_in;
+    unsigned long long bytes_in;
+    unsigned long messages_ignored;
+    unsigned long long bytes_ignored;
+    Duration_t input_processing_time;
+    unsigned long messages_out;
+    unsigned long long bytes_out;
+    unsigned long messages_dropped;
+    unsigned long long bytes_dropped;
+    Duration_t output_processing_time;
+  };
+
   const string HANDLER_STATISTICS_TOPIC_NAME = "Handler Statistics";
 
   @topic
+  @mutable
+  @autoid(HASH)
   struct HandlerStatistics {
     @key GUID_t application_participant_guid;
     @key string name;
     Duration_t interval;
-    unsigned long messages_in;
-    unsigned long long bytes_in;
-    Duration_t rtps_input_processing_time;
-    Duration_t stun_input_processing_time;
-    unsigned long messages_out;
-    unsigned long long bytes_out;
-    unsigned long messages_dropped;
-    unsigned long long bytes_dropped;
-    Duration_t rtps_output_processing_time;
-    Duration_t stun_output_processing_time;
+    ProtocolStatistics rtps;
+    ProtocolStatistics stun;
```

Comment:
I almost did something like this in #2927, but I didn't want to needlessly break any post processing you might have had.